recorder/src/main/java/org/arquillian/cube/q/recorder/TakeNetworkChaosInformation.java

```diff
@@ -0,0 +1,137 @@
+        final ToxiProxyClient.BaseToxic toxic = toxicCreatedEvent.getToxic();
+        final Q.RunCondition runCondition = toxicCreatedEvent.getRunCondition();
+        addToxicInfotoToxics(toxicAction, toxic, runCondition, CREATE);
+    }
+
+    public void captureToxicDetailsAfterUpdate(@Observes ToxicUpdated toxicUpdatedEvent, QNetworkChaosToxic.ToxicAction toxicAction) {
+        final ToxiProxyClient.BaseToxic toxic = toxicUpdatedEvent.getToxic();
+        addToxicInfotoToxics(toxicAction, toxic, null, UPDATE);
+    }
+
+    public void addToxicInfotoToxics(QNetworkChaosToxic.ToxicAction toxicAction, ToxiProxyClient.BaseToxic toxic, Q.RunCondition runCondition, String phase) {
+        final String actionOn = toxicAction.getName();
+        final String toxicType = toxic.getClass().getSimpleName();
+
+        Map<String, Object> toxicInfo = new LinkedHashMap<>();
+        toxicInfo.put("actionon", actionOn);
+        toxicInfo.put("type", toxicType);
+        toxicInfo.put("phase", phase);
+        toxicInfo.put("toxic", toxic);
+        if (runCondition != null) {
+            toxicInfo.put("runcondition", runCondition);
+        }
+
+        toxics.add(toxicInfo);
+    }
+
+
+    public void reportToxicConfiguration(@Observes After event, ReporterConfiguration reporterConfiguration) throws IOException {
+
+        final String fileName = event.getTestMethod().getName() + ".json";
+
+        final FileEntry fileEntry = createFileEntrywithJsonSchema(reporterConfiguration, fileName);
+
+        GroupEntry groupEntry = new GroupEntry();
+        groupEntry.getPropertyEntries().add(fileEntry);
+
+        propertyReportEvent.fire(new PropertyReportEvent(groupEntry));
+        toxics.clear();
+    }
+
+    private FileEntry createFileEntrywithJsonSchema(ReporterConfiguration reporterConfiguration, String fileName) throws IOException {
+        File jsonFile = new File(createDirectoryByNameChaos(reporterConfiguration.getRootDir()), fileName);
+
+        if (!toxics.isEmpty()) {
+            createJsonSchemaAndWriteToFile(jsonFile);
+        }
+
+        final Path rootDir = Paths.get(reporterConfiguration.getRootDir().getName());
+        final Path relativize = rootDir.relativize(jsonFile.toPath());
+
+        FileEntry fileEntry = new FileEntry();
+        fileEntry.setPath(relativize.toString());
+        fileEntry.setType("json");
+
+        return fileEntry;
+    }
+
+    private void createJsonSchemaAndWriteToFile(File file) throws IOException {
+        JsonNodeFactory jsonNodeFactory = new JsonNodeFactory(false);
+        JsonFactory jsonFactory = new JsonFactory();
+        JsonGenerator generator = jsonFactory.createGenerator(file, JsonEncoding.UTF8);
+
+        ObjectMapper mapper = new ObjectMapper();
+        String jsonString = mapper.writeValueAsString(toxics);
+        ArrayNode toxic = (ArrayNode) mapper.readTree(jsonString);
```

Comment:
Is it the only way to capture toxics in the desired structure? We currently serialize, deserialize and seralize again. It might be time consuming.